components/molecule/photoUploader/src/photoTools.js

```diff
@@ -0,0 +1,317 @@
+import {FORM_IMAGE_UPLOADER_DEFAULT_FORMAT_TO_BASE_64_OPTIONS} from './config'
+
+export function formatToBase64({
+  file,
+  url,
+  options = FORM_IMAGE_UPLOADER_DEFAULT_FORMAT_TO_BASE_64_OPTIONS
+}) {
+  if (file) {
+    const reader = new FileReader() //eslint-disable-line
+    reader.readAsDataURL(file)
+    let originalBase64
+    let exifOrientation
+
+    return new Promise((resolve, reject) => {
+      reader.onload = () => {
+        getExifOrientation(file)
+          .then(getExifOrientationResult => {
+            exifOrientation = getExifOrientationResult
+            return resizeImage({
+              base64Image: reader.result,
+              exifOrientation,
+              ...options
+            })
+          })
+          .then(base64Image => {
+            originalBase64 = base64Image
+            return cropAndRotateImage({
+              base64Image,
+              ...options
+            })
+          })
+          .then(value => base64ToBlob(value))
+          .then(({blob, base64}) => {
+            resolve({
+              file,
+              blob,
+              originalBase64,
+              croppedBase64: base64
+            })
+          })
+          .catch(e => {
+            resolve({hasErrors: true})
+          })
+      }
+    })
+  } else {
+    return new Promise((resolve, reject) => {
+      cropAndRotateImage({
+        imageURL: url,
+        ...options
+      })
+        .then(value => base64ToBlob(value))
+        .then(({blob, base64}) => {
+          resolve({
+            url,
+            blob,
+            croppedBase64: base64
+          })
+        })
+        .catch(e => {
+          resolve({url, hasErrors: true})
+        })
+    })
+  }
+}
+
+export function resizeImage({
+  base64Image,
+  exifOrientation,
+  maxImageWidth,
+  maxImageHeight,
+  imageMimeType
+}) {
+  const canvas = document.createElement('canvas')
+  const context = canvas.getContext('2d')
+
+  const image = new Image() // eslint-disable-line
+  image.src = base64Image
+
+  console.log('exif oritentation : ', exifOrientation)
```

Comment:
👀

Reply:
That's the     🚧 Work in progress zone 🚧       👀